src/main/java/org/folio/event/handler/HoldingUpdateAsyncRecordHandler.java

```diff
@@ -122,21 +123,28 @@ private List<String> extractDistinctAdjacentHoldingsToUpdate(HoldingEventHolder
       .toList();
   }
 
-  private Pair<Boolean, List<PoLine>> updatePoLinesInstanceId(HoldingEventHolder holder, List<PoLine> poLines) {
+  private Pair<Boolean, List<PoLine>> updatePoLinesInstance(HoldingEventHolder holder, List<PoLine> poLines) {
     if (holder.instanceIdEqual()) {
-      log.info("updatePoLinesInstanceId:: No instance id was changed (ids are the same), ignoring update");
+      log.info("updatePoLinesInstance:: No instance id was changed (ids are the same), ignoring update");
       return Pair.of(false, List.of());
     }
-    var oldInstanceId = holder.getInstanceIdPair().getLeft();
-    var newInstanceId = holder.getInstanceIdPair().getRight();
+    if (Objects.isNull(holder.getInstance())) {
+      log.warn("updatePoLinesInstance:: Instance is null, instanceId: {}, ignoring update", holder.getInstanceId());
+      return Pair.of(false, List.of());
+    }
+    log.info("updatePoLinesInstance:: Using new instance data for POL update: {}", holder.getInstance().encodePrettily());
```

Comment:
previously we had intitiatives to not log things that can potentially contain some sensitive data, because it's harder to keep logs in secure place then for example DB. Probably to be on more safer side lets do not log full instance object